deepspeed/launcher/launch.py

```diff
@@ -126,6 +139,25 @@ def terminate_process_tree(pid):
         p.kill()
 
 
+from itertools import chain
+
+
+def parse_range(rng):
+    parts = rng.split('-')
+    if 1 > len(parts) > 2:
+        raise ValueError("Bad range: '%s'" % (rng, ))
+    parts = [int(i) for i in parts]
+    start = parts[0]
+    end = start if len(parts) == 1 else parts[1]
+    if start > end:
+        end, start = start, end
+    return range(start, end + 1)
+
+
+def parse_range_list(rngs):
```

Comment:
Would you mind throwing together a unit test for the range list parsing? There are some similar tests in `tests/unit/launcher/test_ds_arguments.py`

Reply:
A test had been added to test_ds_arguments.py to test parse_range_list().  And improve parse_range_list() logic to enforce ascend order and non-overlap.